src/index.ts

```diff
@@ -51,15 +52,26 @@ export default function (
 ): Plugin {
   const filter = createFilter(include, exclude);
 
+  let config: ResolvedConfig
+
   return {
     enforce: 'pre',
     name: 'vite-plugin-glsl',
 
+    configResolved(resolvedConfig) {
+      config = resolvedConfig
+    },
+
     transform (source, shader) {
       if (filter(shader)) {
-        return dataToEsm(loadShaders(
+        const code = new MagicString(dataToEsm(loadShaders(
           source, shader, defaultExtension
-        ));
+        )));
+
+        return {
+          code: code.toString(),
+          map: config.build.sourcemap ? code.generateMap({ hires: true }) : null
+        };
       }
     }
   };
```

Comment:
Alternatively, this can prob just be done with native vite esbuild:

```js
{
  enforce: 'pre',
  name: 'vite-plugin-glsl',

  configResolved(resolvedConfig) {
    config = resolvedConfig
  },

  async transform (source, shader) {
    if (filter(shader)) {
      const code = loadShaders(source, shader, defaultExtension)
      const result = await vite.transformWithEsbuild(code, shader, {
        loader: 'text',
        sourcemap: config.build.sourcemap ? 'external' : null,
      })

      return result;
    }
  }
}
```

Reply:
Great, thanks @CodyJasonBennett , I'll check it in a next few days.